polyfill/index.d.ts

```diff
@@ -0,0 +1,385 @@
+     *   in-range value.
+     * - In `'balance mode'`, out-of-range values are resolved by balancing them
+     *   with the next highest unit.
+     * - In `'reject mode'`, out-of-range values will cause the function to
+     *   throw a RangeError.
+     *
+     * The default is `'constrain'`.
+     */
+    disambiguation: 'constrain' | 'balance' | 'reject';
+    // TODO: should change name from `disambiguation` to `overflow`? See #607.
+    // TODO: should remove `balance` from non-Duration types? See #609.
+  };
+
+  /**
+   * Options for conversions of `Temporal.DateTime` to `Temporal.Absolute`
+   * */
+  export type ToAbsoluteOptions = {
+    /**
+     * Controls handling of invalid or ambiguous times caused by time zone
+     * offset changes like Daylight Saving time (DST) transitions.
+     *
+     * This option is only relevant if a `DateTime` value does not exist in the
+     * destination time zone (e.g. near "Spring Forward" DST transitions), or
+     * exists more than once (e.g. near "Fall Back" DST transitions).
+     *
+     * In case of ambiguous or non-existent times, this option controls what
+     * absolute time to return:
+     * - `'earlier'`: The earlier time of two possible times
+     * - `'later'`: The later of two possible times
+     * - `'reject'`: Throw a RangeError instead.
+     *
+     * The default is `'earlier'`.
+     *
+     * Compatibility Note: the legacy `Date` object (and libraries like
+     * moment.js and Luxon) gives the same result as `earlier` when turning the
+     * clock back, and `later` when setting the clock forward.
+     *
+     * */
+    disambiguation: 'earlier' | 'later' | 'reject';
+    // TODO: should there be a `compatible` (to legacy `Date`, moment, etc.)
+    //       disambiguation option? See #614.
+  };
+
+  /**
+   * Options for arithmetic operations like `plus()` and `minus()`
+   * */
+  export type ArithmeticOptions = {
+    /**
+     * Controls handling of out-of-range arithmetic results.
+     *
+     * If a result is out of range, then `'constrain'` will clamp the result to
+     * the allowed range, while `'reject'` will throw a RangeError.
+     *
+     * The default is `'constrain'`.
+     */
+    disambiguation: 'constrain' | 'reject';
+    // TODO: may change name from `disambiguation` to `overflow`. See #607.
+  };
+
+  /**
+   * Options to control `Duration.prototype.minus()` behavior
+   * */
+  export type DurationMinusOptions = {
+    /**
+     * Controls how to deal with subtractions that result in negative overflows
+     *
+     * In `'balanceConstrain'` mode, negative fields are balanced with the next
+     * highest field so that none of the fields are negative in the result. If
+     * this is not possible, a `RangeError` is thrown.
+     *
+     * In `'balance'` mode, all fields are balanced with the next highest field,
+     * no matter if they are negative or not.
+     *
+     * TODO: add an example of each mode.
```

Comment:
Do these doc comments pop up in the editor if you're using a TypeScript-aware editor? Otherwise we can probably just refer to the JS docs for examples.

Reply:
Yep, that's the main value of doing them in JSDoc. You get the docs inline in the IDE which can be helpful. Here's what it looks like: 

![image](https://user-images.githubusercontent.com/277214/82836658-cd415b80-9e7b-11ea-8347-0bb91a1debbb.png)